File: src/main/java/com/paper/mes/customer/entity/Customer.java
package com.paper.mes.customer.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.paper.mes.common.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

/**
 * 客户档案 sys_customer。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_customer")
public class Customer extends BaseEntity {

    @TableId(type = IdType.ASSIGN_UUID)
    private String uuid;

    private String customerCode;
    private String customerName;
    private String contact;
    private String phone;

    /** 1次结 2月结 */
    private Integer settleType;
    /** 月结对账日 */
    private Integer settleDay;
    /** 锯纸单价 元/刀 */
    private BigDecimal sawPrice;
    /** 复卷单价 元/吨 */
    private BigDecimal rewindPrice;

    /** 1开票 2不开票 */
    private Integer defaultInvoice;
    /** 1含税 2不含税 */
    private Integer priceIncludeTax;
    /** 税率% */
    private BigDecimal taxRate;
    private String taxNo;
    private String invoiceAddress;
    private String bankAccount;
    private String deliveryAddress;
    private Integer customerLevel;
    private String exportTemplate;
    private String remark;
}

File: frontend/src/types/customer.ts
import type { PageQuery } from './common'
import type { ProcessOrderSettlementMode } from './settlementSemantics'
import type {
  CustomerProcessPriceVO,
  CustomerProcessPriceVOBillingBasis,
  CustomerVO,
} from '../api/generated/customerReadOnly'

/** 客户实体，与后端 Customer 对应（含 BaseEntity 通用字段，按需取用）。 */
export type Customer = CustomerVO

export type CustomerProcessPriceBasis = CustomerProcessPriceVOBillingBasis

export type CustomerProcessPrice = CustomerProcessPriceVO

export interface CustomerProcessPriceSaveDTO {
  catalogUuid: string
  billingBasis: CustomerProcessPriceBasis
  price: number
  isDefault: number
}

/** 客户新增/修改入参，与后端 CustomerSaveDTO 对应。 */
export interface CustomerSaveDTO {
  customerCode?: string
  customerName: string
  contact?: string
  phone?: string
  settleType?: ProcessOrderSettlementMode
  settleDay?: number
  sawPrice?: number
  rewindPrice?: number
  defaultInvoice?: number
  priceIncludeTax?: number
  taxRate?: number
  taxNo?: string
  invoiceAddress?: string
  bankAccount?: string
  deliveryAddress?: string
  customerLevel?: number
  exportTemplate?: string
  remark?: string
  processPrices?: CustomerProcessPriceSaveDTO[]
}

/** 客户列表查询入参。 */
export interface CustomerQuery extends PageQuery {
  keyword?: string
}
